membership-attribute-service/app/controllers/AttributeController.scala

```diff
@@ -154,11 +154,19 @@ class AttributeController(
                 logInfoWithCustomFields(s"${user.id} supports the guardian - $attrs - found via $fromWhere", customFields("supporter"))
                 onSuccessSupporter(attrs)
               case None if sendAttributesIfNotFound =>
-                enrichZuoraAttributes(Attributes(user.id), latestOneOffDate, latestMobileSubscription)
+                enrichZuoraAttributes(Attributes(user.id), latestOneOffDate, latestMobileSubscription) // FIXME: This seem like ti should be returning Result not Attributes
               case _ =>
                 onNotFound
             }
-          }
+          }): Future[Result]) /* FIXME: This type annotation should not be necessary. It probably means there is bug above */
+            .recover { case e =>
+              // This branch indicates a serious error to be investigated ASAP, because it likely means we could not
+              // serve from either Zuora or DynamoDB cache. Likely multi-system outage in progress or logic error.
```

Comment:
I think there is a case where you can be a new acquisition so you won't be in the cache yet and then Zuora doesn't respond in time so you don't get anything served

Reply:
I think currently that will respond with 404, because if zuora times out but dynamo does not, then future will still be something like `Future.success(None)`.

Reply:
I think this scenario could be addressed in many cases with a retry of Zuora call. I could put the retry logic in the Zuora http client in membership-common. What do you think?

Reply:
@arausuy https://github.com/guardian/members-data-api/pull/453